### PR TITLE
Run packet-hardware inventorybios during deprov

### PR DESCRIPTION
**Pull request by ScottGarman**
On supported platforms (currently Dell servers), this will output the
state of a handful of BIOS features and their state. For now we'll just
echo the JSON output to get it into the logs. Later on, the data will be
posted to a forthcoming data store.

### CODE REVIEW

**Review by mmlb — DISMISSED** [review]

**Review by markjacksonfishing — DISMISSED** [review]

**Review by markjacksonfishing — DISMISSED** [review]

File: docker/Dockerfile [hunk 1]
```diff
@@ -68,7 +68,15 @@ RUN apt-get update -y && \
     apt-get -qy clean && \
     rm -rf /var/lib/apt/lists/* /tmp/osie/
 
-ARG PACKET_HARDWARE_COMMIT=68a25da2b6a3d3281838bff6b62639438116d0d1
+# ironlib cli wrapper, a prereq for packet-hardware since we don't run it from its container
+COPY lfs/getbiosconfig /tmp/osie/
+RUN cd /tmp/osie && \
+    if [ $(uname -m) = 'x86_64' ]; then \
```
Comment (tstromberg) [inline docker/Dockerfile]: Always quote strings containing variables, command substitutions, spaces or shell meta characters, unless careful unquoted expansion is required or it’s a shell-internal integer 

Stolen from https://google.github.io/styleguide/shellguide.html - but shellcheck also finds this: https://github.com/koalaman/shellcheck/wiki/SC2046

File: docker/scripts/deprovision.sh [hunk 2]
```diff
@@ -64,7 +64,7 @@ function autofail() {
 trap autofail EXIT
 
 # Check BIOS config and update if drift is detected
-if [[ $arch == x86_64 ]] && [[ $reserved != "true" ]]; then
+if [[ $arch == "x86_64" ]] && [[ $reserved != "true" ]]; then
```
Comment (tstromberg) [inline docker/scripts/deprovision.sh:67]: Quote your variables here too to avoid expansion hell.

File: docker/scripts/deprovision.sh [hunk 3]
```diff
@@ -199,7 +199,7 @@ if [[ $preserve_data == false ]]; then
 	# LSI MegaRAID and Dell PERC series 9
 	set_autofail_stage "checking/resetting MegaRAID/PERC RAID controllers"
 	# do not do grep -q, it doesn't play well with pipefail when lots of pci devices exist
-	if lspci -nn | grep -v 'SAS3008' | grep LSI >/dev/null && [[ $arch == x86_64 ]]; then
+	if lspci -nn | grep -v 'SAS3008' | grep LSI >/dev/null && [[ $arch == "x86_64" ]]; then
```
Comment (tstromberg) [inline docker/scripts/deprovision.sh]: Do the cheapest check first -- it isn't necessary to run lspci unless arch is x86_64.

File: docker/scripts/deprovision.sh [hunk 4]
```diff
@@ -213,7 +213,7 @@ if [[ $preserve_data == false ]]; then
 
 	# Marvell (Dell) BOSS-S1
 	set_autofail_stage "checking/resetting Marvell RAID controllers"
-	if lspci -nn | grep '88SE9230' | grep Marvell >/dev/null && [[ $arch == x86_64 ]]; then
+	if lspci -nn | grep '88SE9230' | grep Marvell >/dev/null && [[ $arch == "x86_64" ]]; then
```
Comment (tstromberg) [inline docker/scripts/deprovision.sh]: Check platform first here as well.

File: docker/scripts/deprovision.sh [hunk 5]
```diff
@@ -271,12 +271,21 @@ baremetal_2a2 | baremetal_2a4 | baremetal_hua)
 *)
 	set_autofail_stage "running packet-hardware inventory"
 	packet-hardware inventory --verbose --tinkerbell "${tinkerbell}/hardware-components"
+	if [[ $arch == "x86_64" ]]; then
```
Comment (tstromberg) [inline docker/scripts/deprovision.sh:275]: Can you add a comment as to why the bios is only inventoried for x86-64? It isn't obvious from the context why, so this behavior might get refactored out later.

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 1 above)
Comment (ScottGarman) [reply docker/Dockerfile]: Agreed. You're catching things that are an issue throughout the Dockerfile, so I've updated them everywhere for consistency.

File: docker/scripts/deprovision.sh (hunk 2 above)
Comment (ScottGarman) [reply docker/scripts/deprovision.sh:67]: Done (also fixed throughout the file for consistency). For whatever reason, shellcheck isn't complaining about this, so it's been missed.

File: docker/scripts/deprovision.sh (hunk 3 above)
Comment (ScottGarman) [reply docker/scripts/deprovision.sh]: Nice catch - fixed.

File: docker/scripts/deprovision.sh (hunk 4 above)
Comment (ScottGarman) [reply docker/scripts/deprovision.sh]: Fixed as well.

File: docker/scripts/deprovision.sh (hunk 5 above)
Comment (ScottGarman) [reply docker/scripts/deprovision.sh:275]: Done.

**Review by markjacksonfishing — APPROVED** [review]

**Pull request merged**